Fix directory listing (#513) is not enough

mallowlabs opened the pull request:
The PR #513 is not enough because a problem is remaining.

If the server running path contains whitespaces (or other special characters like #), the directory check does not work.
### Steps to reproduce

```
$ mkdir /tmp/ninja\ files/ && cd /tmp/ninja\ files/
$ git clone --depth 1 git@github.com:ninjaframework/ninja.git
$ cd ninja
$ mvn test
...
Tests in error:
  testServeStaticDirectory(ninja.AssetsControllerTest)

Tests run: 518, Failures: 0, Errors: 1, Skipped: 0
...
```

This PR fixes these problems.

raphaelbauer commented on ninja-core/src/main/java/ninja/AssetsController.java:
```diff
@@ -127,7 +123,7 @@ private void streamOutUrlEntity(URL url, Context context, Result result) {
         if (url == null) {
             context.finalizeHeadersWithoutFlashAndSessionCookie(Results.notFound());
-        } else if (url.getProtocol().equals("file") && new File(url.getPath()).isDirectory()) {
+        } else if (assetsControllerHelper.isDirectoryURL(url)) {
```
It would be great if you could add a mockito.verify to AssetsController Test that makes sure assetsControllerHelper.isDirectoryURL has been called properly...

raphaelbauer commented:
Looks good! I added one comment to make sure we are really calling the correct method of the injected AssetsControllerHelper... Do you think you can add that test?

mallowlabs committed:
Fix directory listing when the server running path contains whitespaces

If Ninja or Tomcat runs on a path contains whiete spaces,
the check for directory listing ( https://github.com/ninjaframework/ninja/pull/513 ) is not enough.
This commit fix this problem.

mallowlabs commented:
@raphaelbauer 
Thank you for your review. I updated my PR!
(CI is failing, but I think it is not caused by my PR.)

raphaelbauer commented on ninja-core/src/test/java/ninja/AssetsControllerTest.java:
```diff
@@ -148,6 +156,7 @@ public void testServeStaticDirectory() throws Exception {
 
         renderable.render(contextRenderable, result);
 
+        verify(assetsControllerHelper).isDirectoryURL(this.getClass().getResource("/assets/"));
```
yea!

raphaelbauer commented:
Thanks! Merged 👍

mallowlabs commented:
Thank you!!
